### PR TITLE
feat(SecretInput):handle null optionalvalue gracefully

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes
Ensure that the `SecretInput` component properly handles null values for the `optionalvalue` prop by providing fallbacks, preventing React warnings about passing null values to the input element's value prop.

### CODE REVIEW

File: packages/webapp/src/components/ui/input/SecretInput.tsx [hunk 1]
```diff
@@ -13,13 +13,17 @@ type SecretInputProps = Omit<JSX.IntrinsicElements['input'], 'defaultValue'> & {
...
+) {
     const [isSecretVisible, setIsSecretVisible] = useState(false);
 
-    const [changedValue, setChangedValue] = useState(props.defaultValue);
+    const [changedValue, setChangedValue] = useState(defaultValue);
 
-    const value = props.optionalvalue === null ? '' : props.optionalvalue || changedValue;
-    const updateValue = props.setoptionalvalue || setChangedValue;
+    // Handle null value in optionalvalue
+    const value = optionalvalue ?? changedValue;
```
Comment (bodinsamuel) [inline packages/webapp/src/components/ui/input/SecretInput.tsx]: SecretInputProps never define `null` as accepted value, I wonder where it was happening and if it's allowed can you modify this type? (and maybe the other type too)

Comment (hassan254-prog) [reply packages/webapp/src/components/ui/input/SecretInput.tsx]: Just updated the data types for these. Thanks @bodinsamuel for this.

Comment (bodinsamuel) [reply packages/webapp/src/components/ui/input/SecretInput.tsx]: I still can't pass `optionalValue={null}` => `Type 'null' is not assignable to type 'string | undefined'.`

Comment (hassan254-prog) [reply packages/webapp/src/components/ui/input/SecretInput.tsx]: Should now work ☺️.Thanks

**Review by bodinsamuel — APPROVED** [review]
👌🏻

**Pull request merged**